factories/esSearcherFactory.js

```diff
@@ -170,6 +170,10 @@
           var doc = parseDoc(hit);
           thisSearcher.docs.push(doc);
         });
+
+        if ( angular.isDefined(data._shards) && data._shards.failed > 0 ) {
```

Comment:
I'm gonna bug you for a test for this. Only because when I did the transport work last week, I had an issue where I got nearly everything working (go tests!). But I wasn't handling errors correctly cause there was no test for that and I nearly shipped it :)